src/sortby.ts

```diff
@@ -0,0 +1,95 @@
+/**
+  * Sort array by a specific element property, its value type must be one of string, number and date
+  * @param {string | undefined} sortByProperty specify a property from each element that sorting will be based on, undefined means sorty by element itself
+  * @param {
+      isDescending: boolean;
+      sortByPropertyType: string;
+    } options 
+  * @returns script version
+  */
+export function SortBy<T>(sortByProperty: string | symbol | number, options: {
+  isDescending: boolean;
+  type: string;
+} = {
+  isDescending: true,
+  type: 'string'
+}) {
+  const cachedValueKey = Symbol();
+  return function(
+    target: any,
+    propertyKey: string,
+    descriptor?: PropertyDescriptor
+  ) {
+    Object.defineProperty(target, propertyKey, {
+      set: function(value: Array<T>) {
+        
+        if (!value || !Array.isArray(value)) {
+          throw `Value of property ${propertyKey} is not a valid array!`;
+        }
+
+        if (!!sortByProperty && value.some(item => !item.hasOwnProperty(sortByProperty))) {
+          throw `Property ${propertyKey} has some elements that do not have property ${sortByProperty ? sortByProperty.toString() : ''}!`;
+        }
+      
+  
+        const errorMsg = (item: any, type: string) => `Value ${item} is not of predefined type ${type}`;
```

Comment:
Move this function to the file level, it's static, no point of recreating it every time we use the decorator.

Reply:
I think it makes a lot of sense and this is my learning from this pull request. Thank you!